### ISSUE TITLE
Support content negotiation for the different license list formats on spdx.org/licenses

### ISSUE THREAD

**Issue by goneall** [spdx/license-list-XML#1166]
The [SPDX license list website](https://spdx.org/licenses/) currently serves JSON, RDFa and HTML formats for the licenses.  JSON is referenced by replacing the `.html` with  `.json`  in the license URL.

A possibly better option to service the different formats for the license list would be to support content negotiation on the website.

See https://developer.mozilla.org/en-US/docs/Web/HTTP/Content_negotiation for an overview of content negotiation.

In addition to JSON and HTML, we could also support the other formats currently available in the [license-list-data](https://github.com/spdx/license-list-data) repo.

This request was raised in the SPDX tech call on 5 Jan 2020.

**Comment by jlovejoy** [comment]
@goneall - I'm not sure what "content negotiation" means or what this entails (is "technical issue" the right label).

Looks like it was raised some time ago on a tech call - something to raise on a legal call or joint call??

**Comment by goneall** [comment]
@jlovejoy This is more of a website configuration issue.  I don't expect it to be very contentious, so we may not need a call.  Just someone knowledgeable enough to implement it (which isn't me ;)  We could ping the tech and legal mailing to see if someone's willing to help.

@swinslow - Is the current website using a standard AWS S3 bucket to store flat HTML/other files?  Do you have or know who has admin access to the site?

I haven't read this, but a quick google search found this article: https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/RequestAndResponseBehaviorS3Origin.html

**Comment by swinslow** [comment]
Hi @goneall, to my knowledge it is a standard AWS S3 bucket. I didn't set it up but I do have at least some level of access to it (enough to be able to log into the console, and also to push the files for each release).

I'm not sure I understand the request well enough to give a meaningful response :) but I'd be happy to jump on a call and look at the details together based on what I've got access to log into.

**Comment by seabass-labrax** [comment]
@jlovejoy, @swinslow - content negotiation is where the web browser or other web client asks to receive the data at a URL in a particular format; the preference being stated in the HTTP header section itself. This is as opposed to having the different formats on different URLs, which is what we are currently doing. Content negotiation would be useful mainly for people who are writing programs to automatically fetch the license data.

@goneall, I think it's a good idea to support content negotiation, but would we continue to support the separate format URLs for backwards compatibility and more human-friendly access? I think it'd be a shame to lose the format-specific URLs, at least for the existing formats.

**Comment by goneall** [comment]
> I think it'd be a shame to lose the format-specific URLs, at least for the existing formats.

I agree, we should keep those.  This would only be in addition to the separate URL's.

I'll do some research on configuring AWS for content negotiation and see how much of a challenge it would be.

**Comment by goneall** [comment]
I found this rather useful [article](https://std.in/http-content-negotiation-on-aws-cloudfront/) on enabling content negotiation for S3.

We would basically need to  configure a CloudFront CDN and the add some JavaScript code.  The JavaScript code looks pretty straight forward.

@swinslow Do you think you have enough permissions to add and configure CloudFront?  If not, we would need to get the LF IT group involved to help enable this.

**Comment by swinslow** [comment]
Hi @goneall, apologies for the slow delay on this. I'm looking into the permissions I've got to see whether it looks like I can add and configure this. My guess is "probably not" but I'm checking.

**Comment by swinslow** [comment]
Okay, yeah... I'm getting a lot of "not authorized to perform X" sorts of errors when trying to access / list or create CloudFront resources. So it looks like we'd need to loop in LF IT to move this forward. @goneall would you like me to reach out and see if I can make the connection there?

**Comment by goneall** [comment]
>  @goneall would you like me to reach out and see if I can make the connection there?

@swinslow that would be great - I'm available to help if needed

**Comment by swinslow** [comment]
Unfortunately I haven't had a chance to engage with LF IT to explore getting AWS permissions to investigate this. Removing 3.19 and shifting to 3.20, will try to dig deeper as part of the next release.
